bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/wrappers/veneer/BigtableHBaseVeneerSettings.java

```diff
@@ -0,0 +1,594 @@
+
+      int maxInflightRpcCount = Integer.parseInt(maxInflightRpcStr);
+
+      // TODO: This needs to be extracted for calculating outstandingElementCount
+      long bulkMaxRowKeyCount =
+          builder.bulkMutateRowsSettings().getBatchingSettings().getElementCountThreshold();
+
+      FlowControlSettings.Builder flowControlBuilder =
+          FlowControlSettings.newBuilder()
+              // TODO: either deprecate maxInflightRpcCount and expose the max outstanding elements
+              // in user configuration or introduce maxInflightRpcCount to gax
+              .setMaxOutstandingElementCount(maxInflightRpcCount * bulkMaxRowKeyCount);
+
+      String maxMemory = configuration.get(BIGTABLE_BUFFERED_MUTATOR_MAX_MEMORY_KEY);
+      if (!isNullOrEmpty(maxMemory)) {
+        flowControlBuilder.setMaxOutstandingRequestBytes(Long.valueOf(maxMemory));
+      }
+
+      batchingSettingsBuilder.setFlowControlSettings(flowControlBuilder.build());
+    }
+
+    String requestByteThresholdStr = configuration.get(BIGTABLE_BULK_MAX_REQUEST_SIZE_BYTES);
+    if (!isNullOrEmpty(requestByteThresholdStr)) {
+      batchingSettingsBuilder.setRequestByteThreshold(Long.valueOf(requestByteThresholdStr));
+    }
+
+    builder.bulkMutateRowsSettings().setBatchingSettings(batchingSettingsBuilder.build());
+
+    configureIdempotentCallSettings(builder.bulkMutateRowsSettings());
+  }
+
+  private void configureBulkReadRowsSettings(EnhancedBigtableStubSettings.Builder builder) {
+    BatchingSettings.Builder bulkReadBatchingBuilder =
+        builder.bulkReadRowsSettings().getBatchingSettings().toBuilder();
+
+    String bulkMaxRowKeyCountStr = configuration.get(BIGTABLE_BULK_MAX_ROW_KEY_COUNT);
+    if (!isNullOrEmpty(bulkMaxRowKeyCountStr)) {
+      bulkReadBatchingBuilder.setElementCountThreshold(Long.valueOf(bulkMaxRowKeyCountStr));
+    }
+
+    builder.bulkReadRowsSettings().setBatchingSettings(bulkReadBatchingBuilder.build());
+
+    configureIdempotentCallSettings(builder.bulkReadRowsSettings());
+  }
+
+  private void configureReadRowsSettings(EnhancedBigtableStubSettings.Builder stubSettings) {
+    RetrySettings.Builder retryBuilder =
+        stubSettings.readRowsSettings().getRetrySettings().toBuilder();
+
+    if (isRetriesDisabled()) {
+
+      stubSettings.readRowsSettings().setRetryableCodes(Collections.<StatusCode.Code>emptySet());
+    } else {
+      ImmutableSet.Builder<StatusCode.Code> retryCodes = ImmutableSet.builder();
+
+      retryCodes
+          .addAll(extractRetryCodesFromConfig())
+          .addAll(stubSettings.readRowsSettings().getRetryableCodes());
+
+      stubSettings.readRowsSettings().setRetryableCodes(retryCodes.build());
+
+      String initialElapsedBackoffMsStr = configuration.get(INITIAL_ELAPSED_BACKOFF_MILLIS_KEY);
+      if (!isNullOrEmpty(initialElapsedBackoffMsStr)) {
+        retryBuilder.setInitialRetryDelay(ofMillis(Long.parseLong(initialElapsedBackoffMsStr)));
```

Comment:
Please check if initialRetryDelay > default maxRetryDelay. It it is, then set the max delay to initial? with a todo fix it in the future

Reply:
Thanks again for the suggestion!

Have added this check to make sure initialRetryDelay & maxRetryDelay is same,
```java
  if (initialElapsedBackoffMs > retryBuilder.getMaxRetryDelay().toMillis()) {
    // TODO: fix this scenario by maybe introducing maxRetryDelayMillis directly
    retryBuilder.setMaxRetryDelay(ofMillis(initialElapsedBackoffMs));
  }
```